### PR TITLE
Changed watermark from GUI to mesh.

### PR DESCRIPTION
**Pull request by asunawu**
Fixed [AB#1031](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/1031) and there was a watermark on the screenshot image too.

### CODE REVIEW

File: src/vcRender.cpp [hunk 1]
```diff
@@ -1130,7 +1149,42 @@ bool vcRender_CreateSelectionBuffer(vcState *pProgramState, vcRenderContext *pRe
   }
   return true;
 }
+void vcRenderWatermark(vcState *pProgramState, vcRenderContext *pRenderContext)
```
Comment (samsinsane) [inline src/vcRender.cpp]: There should be an empty line before this line - functions are separated by empty lines.

File: src/vcRender.cpp [hunk 2]
```diff
@@ -1130,7 +1149,42 @@ bool vcRender_CreateSelectionBuffer(vcState *pProgramState, vcRenderContext *pRe
...
+  vcShader_BindTexture(pRenderContext->watermarkShader.pProgram
+    , pProgramState->pSceneWatermark
+    , 0
+    , pRenderContext->watermarkShader.uniform_texture);
+
+  vcMesh_Render(gInternalMeshes[vcInternalMeshType_Billboard]);
+
+  vcGLState_ResetState();
+  vcShader_Bind(nullptr);
+}
```
Comment (samsinsane) [inline src/vcRender.cpp]: There should be an empty line after this - same reasoning as above.

File: src/vcRender.cpp [hunk 3]
```diff
@@ -1130,7 +1149,42 @@ bool vcRender_CreateSelectionBuffer(vcState *pProgramState, vcRenderContext *pRe
...
+    return;
 
+  vcGLState_SetDepthStencilMode(vcGLSDM_Always, false);
+  vcGLState_SetFaceMode(vcGLSFM_Solid, vcGLSCM_None);
+
+  udInt2 imageSize = udInt2::zero();
+  vcTexture_GetSize(pProgramState->pSceneWatermark, &imageSize.x, &imageSize.y);
+
+  udDouble3 position = udDouble3::create((double)imageSize.x / pRenderContext->sceneResolution.x - 1,
+    (double)imageSize.y / pRenderContext->sceneResolution.y - 1, 0);
```
Comment (samsinsane) [inline src/vcRender.cpp]: These should all be on one line (there's a few more cases below where you do this too).

Also, the udMath types allow you to do something like:
```c++
udDouble3 position = udDouble3::create(imageSize) / pRenderContext->sceneResolution - 1;
```
I'm not 100% certain if the create functions allow for different types - something that we should definitely take care of outside of this PR if they don't.

Comment (asunawu) [reply src/vcRender.cpp]: Sorry, but pRenderContext->sceneResolution's type is udUint2, and 1 should be(1,1,0). Do I need to fix it still?

File: src/vcRender.cpp (hunk 2 above)
Comment (asunawu) [reply src/vcRender.cpp]: Fixed.

File: src/vcRender.cpp (hunk 1 above)
Comment (asunawu) [reply src/vcRender.cpp]: Fixed.

File: src/vcRender.cpp (hunk 3 above)
Comment (samsinsane) [reply src/vcRender.cpp]: Ah right, sorry. You can still utilise the various operators and create functions, something like:
```c++
udDouble2 imageSizeD = udDouble2::create(imageSize);
udDouble2 sceneResD = udDouble2::create(pRenderContext->sceneResolution);
udDouble3 position = udDouble3::create(imageSizeD / sceneResD - 1, 0);
```
You can also get rid of the local variables, but I think you have something similar below too?

Comment (asunawu) [reply src/vcRender.cpp]: udDouble3::create(imageSizeD / sceneResD - 1, 0) won't work.

Comment (samsinsane) [reply src/vcRender.cpp]: Is it because the constants aren't doubles, or have I forgotten something?

Comment (samsinsane) [reply src/vcRender.cpp]: Oh I see, I'm sorry, we have the `*` and `/` operators for a scalar, but not `+` and `-`, I think you would just need to change `1` to `udDouble2::one()`

Comment (asunawu) [reply src/vcRender.cpp]: Do you think it's really worth to do that? The requested change will will add at least 8 additional 4-byte copy operations. I have changed the code into one line for easier reading.

Comment (samsinsane) [reply src/vcRender.cpp]: I was only considering the readability of the code, but you raise a very good point that I was not even thinking about - this is a very bad place to add unnecessary work, oops. I'm going to resolve this discussion. 👍

**Pull request merged**